include/itkAttenuationImageFilter.hxx

```diff
@@ -148,6 +179,12 @@ AttenuationImageFilter<TInputImage, TOutputImage, TMaskImage>::ThreadedGenerateD
 
   const float scanStepMM = input->GetSpacing()[m_Direction];
 
+  thread_local std::vector<float> accumulatedWeight;
+  accumulatedWeight.resize(regionForThread.GetSize(m_Direction));
```

Comment:
Is there an advantage to using a thread-local vector to represent the scan line as opposed to an ITK image like we do for the attenuation output image?

Reply:
This is a temporary variable. We don't want to allocate another image worth of memory for holding this, when we can get away with `xSize`*`numberOfThreads` instead.